theseus/labs/lie/functional/so3_impl.py

```diff
@@ -613,13 +621,14 @@ def _jcompose_impl(
 ) -> Tuple[List[torch.Tensor], torch.Tensor]:
     check_group_tensor(group0)
     check_group_tensor(group1)
-    jacobians = []
-    jacobians.append(group1.transpose(1, 2))
-    jacobians.append(group0.new_zeros(group0.shape[0], 3, 3))
-    jacobians[1][:, 0, 0] = 1
-    jacobians[1][:, 1, 1] = 1
-    jacobians[1][:, 2, 2] = 1
-    return jacobians, group0 @ group1
+    ret = group0 @ group1
+    size = get_group_size(ret)
+    jac0 = group1.transpose(-1, -2).expand(*size, 3, 3)
+    jac1 = group0.new_zeros(*size, 3, 3)
+    jac1[..., 0, 0] = 1
```

Comment:
Maybe something similar to what I proposed for SE3?

Reply:
Fixed